modules/ding_mobilesearch/includes/admin.inc

```diff
@@ -0,0 +1,264 @@
+      db_delete('flag_types')
+        ->condition('fid', $flag->fid)
+        ->execute();
+      foreach ($types as $machine_name => $row) {
+        if ($row['type']) {
+          db_insert('flag_types')
+            ->fields(array(
+              'fid' => $flag->fid,
+              'type' => $machine_name,
+            ))
+            ->execute();
+        }
+      }
+    }
+  }
+
+  // Save CT's settings.
+  foreach ($types as $machine_name => $row) {
+    $key = 'mobilesearch_type__' . $machine_name;
+    variable_set($key, $row['type']);
+    variable_set($key . '__trigger', $row['trigger']);
+    variable_set($key . '__plugin', $row['plugin']);
+  }
+
+  // Save menus settings.
+  $menus = $form_state['values']['menus']['rows'];
+  foreach ($menus as $machine_name => $row) {
+    $key = 'mobilesearch_menu__' . $machine_name;
+    variable_set($key, $row['menu']);
+    variable_set($key . '__plugin', $row['plugin']);
+  }
+
+  drupal_set_message(t('The configuration options have been saved.'));
+}
+
+/**
+ * Returns HTML for the content types exporting form.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - form: A render element representing the form.
+ *
+ * @ingroup themeable
+ *
+ * @return string
+ *   Rendered table.
+ */
+function theme_ding_mobilesearch_types_table(array $variables) {
+  $form = $variables['form'];
+  $header = array(
+    t('Content type'),
+    t('Trigger'),
+    t('Export plugin'),
+  );
+
+  $rows = array();
+  foreach (element_children($form['rows']) as $key) {
+    $type = drupal_render($form['rows'][$key]['type']);
+    $trigger = drupal_render($form['rows'][$key]['trigger']);
+    $plugin = drupal_render($form['rows'][$key]['plugin']);
+    $row = array();
+    $row[] = $type;
+    $row[] = $trigger;
+    $row[] = $plugin;
+    $rows[] = $row;
+  }
+
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+    'empty' => t('No content types available.'),
+  ));
+  $output .= drupal_render_children($form);
+
+  return $output;
+}
+
+/**
+ * Returns HTML for the content types exporting form.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - form: A render element representing the form.
+ *
+ * @ingroup themeable
+ *
+ * @return string
+ *   Rendered table.
+ */
+function theme_ding_mobilesearch_menus_table(array $variables) {
+  $form = $variables['form'];
+  $header = array(
+    t('Menu'),
+    t('Export plugin'),
+  );
+
+  $rows = array();
+  foreach (element_children($form['rows']) as $key) {
+    $menu = drupal_render($form['rows'][$key]['menu']);
+    $plugin = drupal_render($form['rows'][$key]['plugin']);
+    $row = array();
+    $row[] = $menu;
+    $row[] = $plugin;
+    $rows[] = $row;
+  }
+
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+    'empty' => t('No menus available.'),
+  ));
+  $output .= drupal_render_children($form);
+
+  return $output;
```

Comment:
I might be missing something in relation to styling forms, but why not return an render array?